beacon/chain.go

```diff
@@ -213,7 +213,8 @@ func (c *chainStore) runChainLoop() {
 }
 
 func isAppendable(lastBeacon, newBeacon *Beacon) bool {
-	return newBeacon.Round == lastBeacon.Round+1
+	return newBeacon.Round == lastBeacon.Round+1 &&
+		bytes.Equal(lastBeacon.Signature, newBeacon.PreviousSig)
```

Comment:
(1) Here we want to make extra sure the chain is correctly formed. Even though there are already checks before, this is the last one before inserting the beacon in the chain so better add it here to make it explicit :) 
(2) On your comment I think it relates more to Filecoin. Filecoin nodes can do whatever they want they dont need to store the `PreviousSig` since a previous block in Filecoin already stores it. In Drand I'd like to keep the fact that any given beacon is verifiable by itself, you dont need to crawl a chain to use it, only need the distributed public key. Given the relative low size of a beacon (< 200 bytes) it's not much of a big deal.